### DISCUSSION TITLE
Depth issues about rex terrain engine

### DISCUSSION THREAD

**Discussion by Hypo-Li** [pelicanmapping/osgearth#2391] [Support Q&A]
Hello everyone, I am trying to develop an atmosphere for osgEarth based on the UE atmospheric system. This system relies on depth buffering, which obtains depth and calculates the world coordinates to determine distance, and calculates the atmospheric scattering value at that point. However, I have noticed that when using the Rex terrain engine, the depth of the Earth is always concentrated within a range close to 1, which leads to jagged depth buffering When using the MP terrain engine, the depth of the Earth can be well dispersed within 0-1. I created two identical scenes using osgEarth2.10.1 and set the same viewMatrix and projectionMatrix for the camera, one using the Rex engine and the other using the MP engine. The following are the depth buffer textures and depth value distributions captured using Nsight I guess the problem may arise in the calculation of bounding boxes Because I need to implement atmospheric scattering in osgEarth3.4, I would like to know the cause of this problem and whether there is a way to achieve the same depth effect as the mp engine in the rex engine?
![rex](https://github.com/gwaldron/osgearth/assets/48642338/e19b0a05-0855-4063-b231-e8a950a87933)
![MP](https://github.com/gwaldron/osgearth/assets/48642338/2a98e1f6-c97e-404e-bd16-9dada3c27038)

**Comment by Hypo-Li** [comment]
![rex](https://github.com/gwaldron/osgearth/assets/48642338/a15fcb90-fa28-4ed3-be62-17816c9c6296)
![MP](https://github.com/gwaldron/osgearth/assets/48642338/61b552c6-887a-4413-89de-ac6106cb6c74)

**Comment by gwaldron** [comment]
It might be the logarithmic depth buffer causing a difference; it is on by default in osgEarth 3.4. You can disable it with the `--nologdepth` flag when running any of the osgEarth viewers.

**Reply by Hypo-Li** [reply]
It seems that the problem is not caused by the logarithmic depth buffer. In osgEarth 3.4, whether I enable or disable it, I always get the same result: depth distribution around 1.0
